component/async/kafka/simple/simple.go

```diff
@@ -135,8 +154,22 @@ func (c *consumer) Consume(ctx context.Context) (<-chan async.Message, <-chan er
 		return nil, nil, errors.New("got 0 partitions")
 	}
 
-	for _, pc := range pcs {
-		go func(consumer sarama.PartitionConsumer) {
+	var wg sync.WaitGroup
+	if c.notif != nil {
+		wg.Add(len(pcs))
+		go func() {
+			// Wait for all the partition consumers to have reached the latest offset before closing the input channel.
```

Comment:
The latest offset can be reached multiple times though? So this implementation only allows to catch the first time end-of-topic is reached.

I cannot think of an use-case where you want to know it every time it happens at the moment.

Reply:
Also: closing a closed channel will panic, so this will panic 2nd time latest offset is reached?

Reply:
> closing a closed channel will panic, so this will panic 2nd time latest offset is reached?

Why would the channel be closed multiple times? The channel is closed once we reach the last offset for all partitions.

Reply:
Also, I've done a change but just to be mega careful about the offset. I had in mind a very unfortunate scenario where for example, a log compaction occurs between the moment we calculated the latest offset and the moment we expect to consume it, we may not receive it anymore. Hence, I changed the check to be `currentOffset >= latestOffset`.